common/network-common/src/main/java/org/apache/spark/network/client/TransportClientFactory.java

```diff
@@ -276,10 +277,19 @@ public void initChannel(SocketChannel ch) {
     // Connect to the remote server
     long preConnect = System.nanoTime();
     ChannelFuture cf = bootstrap.connect(address);
-    if (!cf.await(conf.connectionCreationTimeoutMs())) {
+
+    if (connCreateTimeout <= 0) {
+      cf.awaitUninterruptibly();
```

Comment:
I think this should be a plain `await()` instead of `awaitUninterruptibly()`, since the latter might lead to uninterruptible canceled tasks. See https://github.com/apache/spark/pull/16866 for a past fix for a similar issue.

Reply:
Thank you. Could you make a PR to fix it?

Reply:
It seems that we need a new JIRA to fix that instead of a followup.

Reply:
Filed https://issues.apache.org/jira/browse/SPARK-44920 and opened https://github.com/apache/spark/pull/42619 to fix this.

Reply:
Thank you @JoshRosen. The fix looks good to me.

While considering the usage of `await` and `awaitUninterruptibly`, I consulted the Netty documentation (https://netty.io/4.0/api/io/netty/channel/ChannelFuture.html) which recommended the use of `awaitUninterruptibly` in all its GOOD examples. So...